exp/api/v1beta1/types.go

```diff
@@ -37,21 +37,20 @@ type Taint struct {
 }
 
 // ProvisioningState describes the provisioning state of an GCP resource.
+// Ref - https://cloud.google.com/compute/docs/instances/instance-life-cycle
 type ProvisioningState string
 
 const (
-	// Creating ...
-	Creating ProvisioningState = "Creating"
-	// Deleting ...
-	Deleting ProvisioningState = "Deleting"
-	// Failed ...
-	Failed ProvisioningState = "Failed"
-	// Succeeded ...
+	// Provisioning - resources are allocated for the VM. The VM is not running yet
```

Comment:
thinking if we could re-use the status fields here https://github.com/newrelic-forks/cluster-api-provider-gcp/blob/nr-main/api/v1beta1/types.go#L216-L246

Reply:
then we only have to add de-provisioning there if we want that represented.

Reply:
I had initially considered that. However, GCP follows a different convention when it comes to the instance state unlike Azure and AWS. I think it would be best to align with the physical state for better readability.

https://cloud.google.com/compute/docs/instances/instance-life-cycle